src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -336,22 +443,65 @@ private static void ThrowCannotWriteToDeflateStreamException()
+        /// Call this method to determine how many bytes were read from the stream. This method can be called once to return the amount of bytes read between calls to <see cref="System.IO.Compression.DeflateStream.BeginRead" /> and <see cref="System.IO.Compression.DeflateStream.EndRead" />.
+        /// This method blocks until the I/O operation has completed.</remarks>
+        /// <exception cref="System.ArgumentNullException"><paramref name="asyncResult" /> is <see langword="null" />.</exception>
+        /// <exception cref="System.ArgumentException"><paramref name="asyncResult" /> did not originate from a <see cref="System.IO.Compression.DeflateStream.BeginRead(byte[],int,int,System.AsyncCallback,object)" /> method on the current stream.</exception>
+        /// <exception cref="System.SystemException">An exception was thrown during a call to <see cref="O:System.Threading.WaitHandle.WaitOne" />.</exception>
+        /// <exception cref="System.InvalidOperationException">The end call is invalid because asynchronous read operations for this stream are not yet complete.
+        /// -or-
+        /// The stream is <see langword="null" />.</exception>
         public override int EndRead(IAsyncResult asyncResult)
         {
             EnsureDecompressionMode();
             EnsureNotDisposed();
             return TaskToApm.End<int>(asyncResult);
         }
 
+        /// <summary>Asynchronously reads a sequence of bytes from the current Deflate stream, writes them to a byte array, advances the position within the Deflate stream by the number of bytes read, and monitors cancellation requests.</summary>
+        /// <param name="buffer">The buffer to write the data into.</param>
+        /// <param name="offset">The byte offset in <paramref name="buffer" /> at which to begin writing data from the Deflate stream.</param>
+        /// <param name="count">The maximum number of bytes to read.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="System.Threading.CancellationToken.None" />.</param>
+        /// <returns>A task that represents the asynchronous read operation, which wraps the total number of bytes read into the <paramref name="buffer" />. The result value can be less than the number of bytes requested if the number of bytes currently available is less than the requested number, or it can be 0 (zero) if the end of the Deflate stream has been reached.</returns>
+        /// <remarks>The `ReadAsync` method enables you to perform resource-intensive I/O operations without blocking the main thread. This performance consideration is particularly important in a Windows 8.x Store app or desktop app where a time-consuming stream operation can block the UI thread and make your app appear as if it is not working. The async methods are used in conjunction with the <see langword="async" /> and <see langword="await" /> keywords in Visual Basic and C#.
```

Comment:
Need to double check the cref signature in VS:
```suggestion
        /// <remarks>The <see cref="System.IO.Compression.DeflateStream.ReadAsync(System.Byte[],System.Int32,System.Int32,System.Threading.CancellationToken)" /> method enables you to perform resource-intensive I/O operations without blocking the main thread. This performance consideration is particularly important in a Windows 8.x Store app or desktop app where a time-consuming stream operation can block the UI thread and make your app appear as if it is not working. The asynchronous methods are used in conjunction with the <see langword="async" /> and <see langword="await" /> keywords in Visual Basic and C#.
```